astropy/io/fits/convenience.py

```diff
@@ -433,15 +434,24 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
-def table_to_hdu(table):
+def table_to_hdu(table, astropy_native=False):
```

Comment:
@taldcroft - note the difference between reading (where I agree we should default to `False`), and writing, where there is no backward compatibility issue, since before `Time` could not be written at all.  And I really dislike writing something that depends on the arbitrary format someone happened to set! 

However, if the default is that we simply continue to error on mixin columns with `astropy_native=False`, then I'm fine with that (but this is *not* what is currently implemented!)